<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>PRIDE Utilities</title>
    
    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    
    <script src="http://ajax.microsoft.com/ajax/jquery/jquery-1.4.2.min.js" type="text/javascript"></script>
    <script src="stylesheets/javascripts/github.js" type="text/javascript"></script>
    
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1 id="logo" class="logo"><a href="http://pride-utilities.github.io">PRIDE Utilities</a></h1>
        <p>Libraries and Algorithms supporting the PRIDE Database ToolSuite</p>
        <p class="view"><a href="http://github.com/PRIDE-Utilities">GitHub <small>Pride utilities related libraries</small></a></p>
        <p><h4>List of Related Publications:</h4></p>
        <p class="view"><a href="http://dx.doi.org/10.1038/nbt.2112">Nat Biotechnol. 2012 Feb 8;30(2):135-7.<small>Wang R, et al.</small></p>
        <p class="view"><a href="http://www.sciencedirect.com/science/article/pii/S1570963913001039">Biochim Biophys Acta. 2014 Jan;1844(1 Pt A):63-76.<small>Perez-Riverol, et al.</small></p>
      </header>
      <section>
        <h1>PRIDE Utilities Libraries and Algorithms</h1>
        <p><em>View On <a href="http://github.com/PRIDE-Utilities">source of this content</a>.</em></p>
        <h2 class="view">ms-data-core-api</h2>
        <header>
          <ul>
          <li><a href="https://github.com/PRIDE-Utilities/ms-data-core-api/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/PRIDE-Utilities/ms-data-core-api/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="http://github.com/PRIDE-Utilities/ms-data-core-api/">View On 	<strong>GitHub</strong></a></li>
        </ul>
        </header>
        <p>The <a href="https://github.com/PRIDE-Utilities/ms-data-core-api">ms-data-core-api</a> is a modular and open-source library aimed to develop computational proteomics tools. 
           The API, written in Java, enables rapid tool creation by providing a robust, pluggable programming interface
           and common data model. The data object model is based on controlled vocabularies/ontologies and captures the whole
           range of data types included in common proteomics experimental workflows, going from spectra and identification
           data to quantification results. The library contains readers of three of the most used Proteomics Standards
           Initiative standard file formats (<a href="http://www.psidev.info/mzml_1_0_0%20">mzML</a>, <a href="http://www.psidev.info/mzidentml">mzIdentML</a> and <a href="http://www.psidev.info/mztab">mzTab</a>). 
           In addition to mzML, it also supports the other most commonly used mass spectra formats: dta, ms2, mgf, pkl, apl
           (text-based), mzXML and mzData (XML-based).</p>
           
        <p>Related Documents:</p>
        <ul>
            <li>Project <a href="https://github.com/PRIDE-Utilities/ms-data-core-api/issues">Issues</a></li>
            <li>Project <a href="https://github.com/PRIDE-Utilities/ms-data-core-api/wiki">Wiki</a></li>
            <li><a href="http://pride-utilities.github.io/ms-data-core-api/javaDoc/">JavaDoc</a></li>
        </ul>   
        <p>List of libraries and tools that use <strong>ms-data-core-api</strong></p>
        
        <table>
          <tbody>
             <tr>
              <th>Name</th>
              <th>Repository</th>
              <th>Description</th>
             </tr>
             <tr>
              <td>PRIDE Inspector</td>
              <td><a href="http://github.com/PRIDE-Tooluiste/pride-inspector">pride-inspector</a></td>
              <td>A Desktop visualisation tool for proteomics experiments</td>
             </tr>
             <tr>
              <td>PRIDE protein inference library</td>
              <td><a href="https://github.com/PRIDE-Utilities/pride-protein-inference">pride-protein-inference</a></td>
              <td>A library to compute protein inference in proteomics experiments</td>
             </tr>
              <tr>
              <td>PRIDE Repository Pipelines</td>
              <td><a href="http://www.ebi.ac.uk/pride/archive/"> (PRIDE Database) </a></td>
              <td>The Proteomics Identification Database</td>
             </tr>
        </tbody></table>
        
        <h2 class="view">PRIDE Utilities</h2>
        <header>
          <ul>
          <li><a href="https://github.com/PRIDE-Utilities/pride-utilities/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/PRIDE-Utilities/pride-utilities/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="http://github.com/PRIDE-Utilities/pride-utilities/">View On 	<strong>GitHub</strong></a></li>
        </ul>
        </header>
        <p>The <a href="https://github.com/PRIDE-Utilities/pride-utilities">PRIDE Utilities</a> module contains a series of data structures
           and algorithms used by all the components of the PRIDE Inspector Toolsuite and other PRIDE Projects such as <a href="http://github.com/PRIDE-Archive/">PRIDE Archive libraries</a> 
           and <a href="http://github.com/PRIDE-Cluster/">PRIDE Cluster</a>. Some of the values defined in PRIDE Utilities are the definition
           of the amino acid mass table, pK values and hydrophobic indexes. The module also contains the mappings between different
           ontology terms meaning the same concept, e.g. the b ion annotation could be annotated using the PRIDE ontology term 
           PRIDE:0000194 or the PSI-MS ontology term MS:1001224.  These modules homogenize all the terms and concepts used in metadata 
           annotations. For instance, the library contains the definition of the well-established search engines and processing software and
           their corresponding scores in different controlled vocabularies (CVs) or ontologies. </p>
           
        <p>Related Documents:</p>
        <ul>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-utilities/issues">Issues</a></li>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-utilities/wiki">Wiki</a></li>
            <li><a href="http://github.com/mojombo/god/commit/be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2">JavaDoc</a></li>
        </ul>   
        
        <h2 class="view">PRIDE Protein Inference</h2>
        <header>
          <ul>
          <li><a href="https://github.com/PRIDE-Utilities/pride-protein-inference/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/PRIDE-Utilities/pride-protein-inference/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="http://github.com/PRIDE-Utilities/pride-protein-inferences/">View On 	<strong>GitHub</strong></a></li>
        </ul>
        </header>
        <p><a href="https://github.com/PRIDE-Utilities/pride-protein-inference">Protein Inference Algorithms</a> (PIA) is a toolbox for MS based
           protein inference and identification analysis. PIA allows you to inspect the results of common proteomics spectrum
           identification search engines, combine them and conduct statistical analyses. The main focus of PIA lays on the integrated 
           inference algorithms, i.e. concluding the proteins from a set of identified spectra. But it also allows you to inspect your 
           peptide spectrum matches, calculate FDR values across different search engine results and visualize the correspondence between
           PSMs, peptides and proteins.</p>

         <p> Most search engines for protein identification in MS/MS experiments return protein lists, although the actual search yields
             a set of peptide spectrum matches (PSMs). The step from PSMs to proteins is called “protein inference”. If a set of
             identified PSMs supports the detection of more than one protein in the searched database (“protein ambiguity”), usually only
             one representative accession is reported. These representatives may differ according to the used search engine and settings.
             Thus the protein lists of different search engines generally cannot be compared with one another. PSMs of complementary 
             search engines are often combined to enhance the number of reported proteins or to verify the evidence of a peptide,
            which is improved by detection with distinct algorithms.</p>

        <p>Related Documents:</p>
        <ul>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-protein-inference/issues">Issues</a></li>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-protein-inference/wiki">Wiki</a></li>
            <li><a href="http://github.com/mojombo/god/commit/be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2">JavaDoc</a></li>
        </ul>   
        
        <h2 class="view">PRIDE Modifications</h2>
        <header>
          <ul>
          <li><a href="https://github.com/PRIDE-Utilities/pride-mod/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/PRIDE-Utilities/pride-mod/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="http://github.com/PRIDE-Utilities/pride-mod/">View On 	<strong>GitHub</strong></a></li>
        </ul>
        </header>
        <p>The <a href="https://github.com/PRIDE-Utilities/pride-mod">PRIDE Modification </a> library is used to retrieve the protein
           modification information for a specific identifier from different databases: Unimod, PSI-MOD and the PRIDE Modification 
           controlled vocabulary.</p>
        <p> Protein post-translational modifications (PTM) increase the functional diversity of the proteome by the covalent addition 
            of functional groups or proteins, proteolytic cleavage of regulatory subunits or degradation of entire proteins. These
            modifications include phosphorylation, glycosylation, ubiquitination, nitrosylation, methylation, acetylation, lipidation 
            and proteolysis and influence almost all aspects of normal cell biology and pathogenesis. Therefore, identifying and
            understanding PTMs is critical in the study of cell biology and disease treatment and prevention. In addition to PTMs,
            there are other artefactual protein modifications that are added due to the experimental protocol followed by the researchers.
            Some examples are carbamydomethylation or oxidation. The proteomics community has developed tow major resources for protein 
            modifications (including PTMs): Unimod13 and PSI-MOD14. However, modification idenfifiers from these two resources are not 
            trivial to map since some of the modification in Unimod are not present in PSI-MOD and vice versa. Also, every search 
            engine uses their notation and either Unimod or PSI-MOD. The PRIDE Modification library is used to retrieve the modification
            information for a specific identifier from different databases: Unimod, PSI-MOD and the PRIDE Modification controlled 
            vocabulary (internal nomenclature used in PRIDE tools). This library is now used by different tools and pipelines.
            </p>

        <p>Related Documents:</p>
        <ul>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-mod/issues">Issues</a></li>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-mod/wiki">Wiki</a></li>
            <li><a href="http://github.com/mojombo/god/commit/be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2">JavaDoc</a></li>
        </ul>   
        
        <h2 class="view">PRIDE XML Parser</h2>
        <header>
          <ul>
          <li><a href="https://github.com/PRIDE-Utilities/pride-mod/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/PRIDE-Utilities/pride-mod/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="http://github.com/PRIDE-Utilities/pride-mod/">View On 	<strong>GitHub</strong></a></li>
        </ul>
        </header>
        <p>The <a href="https://github.com/PRIDE-Utilities/pride-jaxb">PRIDE XML JAXB library</a> is a library for indexing and parsing
         PRIDE XML 2.1 files. This library does not load the whole file into the memory up-front, instead it employs XML indexing 
         technique to index the file on the fly which gives you fast access and small meomory footprint. Additionally, all entities
         in PRIDE XML file are mapped as objects, and the internal reference between the objects are resolved automatically, this gives
         you direct access in the object model to entities that are only referenced by ID in the actual XML file.</p>

        <p>Related Documents:</p>
        <ul>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-jaxb/issues">Issues</a></li>
            <li>Project <a href="https://github.com/PRIDE-Utilities/pride-jaxb/wiki">Wiki</a></li>
            <li><a href="http://github.com/mojombo/god/commit/be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2">JavaDoc</a></li>
        </ul> 
        
        <h2 class="view">Protein Details Fetcher</h2>
        <header>
          <ul>
          <li><a href="https://github.com/PRIDE-Utilities/protein-detail-fetcher/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/PRIDE-Utilities/protein-detail-fetcher/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="http://github.com/PRIDE-Utilities/protein-detail-fetcher/">View On 	<strong>GitHub</strong></a></li>
        </ul>
        </header>
        <p>The <a href="https://github.com/PRIDE-Utilities/protein-detail-fetcher">Protein Details Fetcher</a> is a library for indexing
         retrieving Protein Information and Details from different databases such as UNIPROT, ENSEMBL, NCBI, etc. It retrieve the information 
         related with the sequence, the status of the protein. </p>

        <p>Related Documents:</p>
        <ul>
            <li>Project <a href="https://github.com/PRIDE-Utilities/protein-detail-fetcher/issues">Issues</a></li>
            <li>Project <a href="https://github.com/PRIDE-Utilities/protein-detail-fetcher/wiki">Wiki</a></li>
            <li><a href="http://github.com/mojombo/god/commit/be6a8cc1c1ecfe9489fb51e4869af15a13fc2cd2">JavaDoc</a></li>
        </ul> 
        <h2> About Us</h2>
        <ul>
          <li><a href="https://github.com/orgs/PRIDE-Utilities/people">PRIDE Development Team</a></li>
        </ul>
       </section>
      <footer>
        <p>This project is maintained by <a href="mailto:pride-support@ebi.ac.uk">PRIDE Team</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
  </body>
</html>